builtin/rebase--helper.c

```diff
@@ -73,18 +79,59 @@ int cmd_rebase__helper(int argc, const char **argv, const char *prefix)
 	flags |= abbreviate_commands ? TODO_LIST_ABBREVIATE_CMDS : 0;
 	flags |= rebase_merges ? TODO_LIST_REBASE_MERGES : 0;
 	flags |= rebase_cousins > 0 ? TODO_LIST_REBASE_COUSINS : 0;
-	flags |= command == SHORTEN_OIDS ? TODO_LIST_SHORTEN_IDS : 0;
 
 	if (rebase_cousins >= 0 && !rebase_merges)
 		warning(_("--[no-]rebase-cousins has no effect without "
 			  "--rebase-merges"));
 
+	if (command == MAKE_SCRIPT && !upstream && squash_onto) {
+		FILE *squash_onto_file;
+		int ret;
+
+		squash_onto_file = fopen(path_squash_onto(), "w");
+		if (squash_onto_file == NULL)
+			return error_errno(_("could not open '%s'"), path_squash_onto());
+
+		ret = fputs(squash_onto, squash_onto_file);
+		fclose(squash_onto_file);
+
+		if (ret < 0)
+			return error_errno(_("could not write to '%s'"), path_squash_onto());
+	}
+
+	if (command == COMPLETE_ACTION || command == MAKE_SCRIPT) {
+		const char *shortupstream, *shorthead;
+		struct object_id orig_head;
+
+		if (get_oid("HEAD", &orig_head))
+			die(_("No HEAD?"));
+
+		head_hash = find_unique_abbrev(&orig_head, GIT_MAX_HEXSZ);
+		shorthead = find_unique_abbrev(&orig_head, DEFAULT_ABBREV);
+
+		if (upstream) {
+			struct object_id upstream_oid;
+
+			get_oid(upstream, &upstream_oid);
+			shortupstream = find_unique_abbrev(&upstream_oid, DEFAULT_ABBREV);
+			strbuf_addf(&revisions, "%s...%s", upstream, head_hash);
+			strbuf_addf(&shortrevisions, "%s..%s", shortupstream, shorthead);
+		} else {
+			strbuf_addf(&revisions, "%s...%s", onto, head_hash);
+			strbuf_add(&shortrevisions, shorthead, strlen(shorthead));
+		}
+	}
+
 	if (command == CONTINUE && argc == 1)
 		return !!sequencer_continue(&opts);
 	if (command == ABORT && argc == 1)
 		return !!sequencer_remove_state(&opts);
-	if (command == MAKE_SCRIPT && argc > 1)
-		return !!sequencer_make_script(stdout, argc, argv, flags);
+	if (command == MAKE_SCRIPT && (argc == 1 || argc == 2)) {
+		const char *restrict_rev = (argc == 2) ? argv[1] : NULL;
+		const char *f_argv[3] = {argv[0], revisions.buf, restrict_rev};
```

Comment:
That is not really readable. `f_argv`? Shouldn't the `argv` list be `NULL` terminated? I have to admit that it took me a while to wrap my head around this...

Also, unless my memory went bad, we have to add the `restrict_revision` (why change the name? That's unnecessary) with a prefixed `^` to make it negative.

You can avoid all that by using an `argv_array`. Please do so.

Reply:
I’m also dissatisfied with `f_argv`, do you have a better name to suggest?

Reply:
Maybe `rev_list_args` or `rev_args`.